ethcore/src/engines/mod.rs

```diff
@@ -265,6 +266,9 @@ pub trait Engine<M: Machine>: Sync + Send {
 		Ok(())
 	}
 
+	/// Allow returning new block header after seal generation. Currently only used by Clique.
+	fn on_seal_block(&self, _block: &M::LiveBlock) -> Result<Option<M::Header>, M::Error> { Ok(None) }
```

Comment:
What's the reason to use this over creating another return enum type for `Engine::generate_seal`?

Reply:
I would prefer not to modify miner code in that way since I think the correct way forward is to refactor it according to GETH style .